src/dune_rules/lib_file_deps.ml

```diff
@@ -1,22 +1,33 @@
 open Import
 
 module Group = struct
-  type t =
+  type ocaml =
     | Cmi
     | Cmx
+
+  type melange =
```

Comment:
There's already a few instances of this type already. What about introducing a `Melange` module (in dune_rules) that will define common types/facts about melange.

Reply:
I attempted this in https://github.com/jchavarri/dune/pull/4/commits/0cf410efbee293c2acc5e103f52cc8119b678875. I had to move `Lib_mode` to `dune_rules` as well, but I believe that should be fine. Also removed some functions from `Lib_mode` that were not used.

I think code looks better now (more similarities between the way OCaml and Melange modes are handled), wdyt?